pex/tracer.py

```diff
@@ -8,29 +8,36 @@
 import time
 from contextlib import contextmanager
 
+from pex.typing import TYPE_CHECKING, cast
 from pex.variables import ENV
 
 __all__ = ("TraceLogger",)
 
+if TYPE_CHECKING:
+    from typing import Any, Callable, IO, Iterator, List, Optional
+
 
 class Trace(object):
     __slots__ = ("msg", "verbosity", "parent", "children", "_clock", "_start", "_stop")
 
     def __init__(self, msg, parent=None, verbosity=1, clock=time):
+        # type: (str, Optional[Trace], int, Any) -> None
         self.msg = msg
         self.verbosity = verbosity
         self.parent = parent
         if parent is not None:
-            parent.children.append(self)
-        self.children = []
+            parent.children.append(self)  # type: ignore[has-type]
+        self.children = cast("List[Trace]", [])
```

Comment:
This is not idiomatic as I read the mypy docs. You want `var = val  # type: X` for typing variables.

Reply:
This fixed the type ignore. Thanks!